### ISSUE TITLE
Don't discard materials

### ISSUE THREAD

**Issue by stackcollision** [stackcollision/GardenConquest#10]
When a block is removed due to a violation (i.e. player tries to place an armor block on a full grid), spawn the materials used to place it so it's not lost)

**Comment by zrisher** [comment]
When placing, this will always be either a single steel plate, interior plate, or construction component right? So definitely nice, but not a huge priority given the small cost of placing a block.

However, when we're removing fully-constructed blocks that are over a limit, I could see this being a very nice feature.

**Comment by stackcollision** [comment]
That was why I didn't do it originally.  But when you merge two licensed grids, one of their beacons has got to go, and that would destroy ship licenses which would make people very unhappy.

**Comment by zrisher** [comment]
Yep definitely makes sense, think that fits under the latter case. But license blocks might be hidden in the innards of a ship, could make recovering the components in time really difficult especially if the FloatingObject count on the server is low (and most are b/c it's a major performance factor). Maybe we could use the CP reward logic to find a good nonfull cargo container to put them in instead of floating them?

**Comment by stackcollision** [comment]
Stockpile status is locked behind the non-interface version MySlimBlock.  Doesn't seem to be a way to get at it.  This may not be feasible.

**Comment by zrisher** [comment]
Well that's unfortunate but not really surprising, I can think of mods that move things between containers, but nothing that creates components in inventories. I don't think it's too onerous to remove things we've warned users about several times though, especially if we still spawn the components in space.

**Comment by zrisher** [comment]
Opened a PR for the changes required for us to be able to spawn components:
https://github.com/KeenSoftwareHouse/SpaceEngineers/pull/52

**Comment by zrisher** [comment]
Now in master, should be possible to implement after next thursday's patch.
